ranger/ext/popen23.py

```diff
@@ -13,6 +13,14 @@
     from sys import version_info
     PY3 = version_info[0] >= 3
 
+if PY3:
+    # pylint: disable=ungrouped-imports,unused-import
+    from subprocess import DEVNULL
+else:
+    import os
+    # pylint: disable=consider-using-with
+    DEVNULL = open(os.devnull, "wb")
```

Comment:
Just realized that this won't work. The first time you use this in a Popen call the file object will be closed once the process finishes, no?

@remi6397, thoughts on a better fix? Maybe we should incorporate the behavior into Popen23 and exclusively use that to interface with Popen?

Reply:
It would only be closed if we called `communicate` on the Popen object, so I believe that it's not going to cause any trouble (for now).

Reply:
I don't really like leaving mines around though.

Any thoughts on a good way forward?

Reply:
Hmmm… Perhaps

```
DEVNULL.close = lambda: None
```

would work? 😅

Reply:
I don't think it would. DEVNULL gets passed by reference as `stdout` or `stderr` and then close is called on those, no?

Reply:
Close would be called on what exactly? I don't really understand… kwargs don't wrap objects and as you said stuff is passed by reference.

Reply:
That could be Python 2-specific, as it works for me on 3.12. If that's the case, then the most reasonable solution would be to drop support for decades old Python versions…

Reply:
Adding DEVNULL as an open file object is specific to Python 2. You guarded it behind the `PY3` flag yourself.

So long as we don't get a significant benefit from dropping Python 2 support it's a shame to give up on it for what's essentially a paper cut.

Reply:
I guess if my solution does indeed not work with ancient Python (I don't really have a way to check that, because Python 2 doesn't build properly on my machine) we could make a dedicated "virtual" file-like object. Something like this perhaps?

```python
class DevNull:
    def __init__(self):
        self._reset()
    def _reset(self):
        self.__f = open(os.devnull, "wb")
    def __getattr__(self, attr):
        if self.__f.closed:
            self._reset()
        return getattr(self.__f, attr)

DEVNULL = DevNull()
```

Reply:
That looks good to me.